### ISSUE TITLE
Create a meeting time report

### ISSUE THREAD

**Issue by TiagoPaisAlmeida** [AntonyCorbett/OnlyT#15]

**Comment by AntonyCorbett** [comment]
Similar to the PDF report found in SoundBox - a detailed record of the meeting talk times and a summary showing aveerage times by week.

**Comment by AndreRobinson** [comment]
I wanted to know how this is accomplished? I looked everywhere and have seen no setting about creating a time report. I only see the folder logs under documents/ Only T. Am I missing something?

Thanks

**Comment by AntonyCorbett** [comment]
@AndreRobinson Hi. The "Issues" section contains a list of bugs, feature requests ("enhancements"), etc. This item describes a feature that has been requested; it's not yet implemented.

**Comment by AndreRobinson** [comment]
Oic sorry for the confusion. Thanks for the quick reply!

Yb
Andre

**Comment by bmanx2** [comment]
This feature is something we're eagerly anticipating here. This is really the only thing that's keeping us from transitioning from the SoundBox timer.

**Comment by Blann** [comment]
Just a quick note that this can (after a fashion) be done using the OnlyT remote app: 
https://onlyt.app/

**Comment by timjobutler** [comment]
Yes, this feature is very much missed since migrating from SoundBox. It's implementation is highly anticipated.

**Comment by AntonyCorbett** [comment]
@timjobutler Sorry, there's no timerline ATM. Shortly I will analyse what requests ther are and try to prioritise

**Comment by AntonyCorbett** [comment]
@timjobutler @bmanx2 @AndreRobinson This feature should be ready to test shortly

**Comment by timjobutler** [comment]
@AntonyCorbett I look forward to testing this feature. Thank you for the work you have put into it.

**Comment by AntonyCorbett** [comment]
@timjobutler Please see latest pre-release

**Comment by LvDaniel** [comment]
Unfortunately the link does't work for downloading OnlyT 1.1.0.50

**Comment by bergerchemical** [comment]
Works, I just checked.  
I also checked the report generator. A folder is created but there are no records.

**Comment by LvDaniel** [comment]
Yep, now it works, thanks.
I'll check the report generator and provide feedback.

**Comment by AntonyCorbett** [comment]
@bergerchemical The report is only generated when OnlyT detects a full meeting agenda. Various heuristics are used to ensure that any experimental or accidental use of the timer does not generate gratuitous reports. This means that if you want to test it you will need to operate it in real time for ~105 minutes.

**Comment by richfrantz** [comment]
That explains why I didn't get a report either. Thanks.
________________________________

**Comment by LvDaniel** [comment]
The message that the report is generating confused me as well. So we need to test in real time then.
Anthony, is it possible to get the message only when the report is actually generated?

**Comment by LvDaniel** [comment]
And also to have the possibility to change the timing folder?

**Comment by AntonyCorbett** [comment]
@LvDaniel The message was meant to alert the users that there is some activity during which time they shouldn't close the app. In normal circumstances the report will be generated. I may need to rethink.

**Comment by AntonyCorbett** [comment]
@LvDaniel Please create separate issues as necessary to raise requests, etc. Thanks for valuable feedback!

**Comment by LvDaniel** [comment]
Sure Anthony, I will.
And a big thanks for all your efforts you put in helping others.

**Comment by LvDaniel** [comment]
I've just tested this in real time. Actually the meeting exceeded 13 minutes and the report was not created due to duration being out of range.
2018-11-13 20:23:11.608 +02:00 [Information] ==== Launched ====
2018-11-13 20:23:11.639 +02:00 [Information] Version 1.1.0.50
....................
2018-11-13 22:21:13.457 +02:00 [Warning] Meeting duration is out of range (26.3 overtime)
2018-11-13 22:21:13.458 +02:00 [Warning] Meeting times invalid so not stored
Should't the range be more flexible? Let's say to be between 30min and 2hrs... just a suggestion.

**Comment by AntonyCorbett** [comment]
@LvDaniel If the recorded meeting duration is more than 20 minutes longer than it should be (105 mins), OnlyV assumes that the data is not a valid cong meeting and ignores it. I have copied the code from SoundBox implementation so it's been this way for a couple of years,

**Comment by bergerchemical** [comment]
@AntonyCorbett My report was not created. In the settings I chose the automatic mode. I'll check it again

**Comment by AntonyCorbett** [comment]
@bergerchemical the log file should indicate why not.

**Comment by LvDaniel** [comment]
Anthony, does it matter if I run the countdown before starting the meeting (and the meeting start time is specified) or this doesn't have to do with the report?
I'll test once again, let's see how it goes.

**Comment by AntonyCorbett** [comment]
@LvDaniel the version you have ignores the countdown and assumes that a meeting starts on the nearest 15 minute interval to the start of the first timer.actually it takes 5 mins off the start of the first timer and uses the nearest quarter if an hour as mtg start

**Comment by AntonyCorbett** [comment]
@bergerchemical From the wiki: OnlyT maintains a log (a simple text file) of activity which can be useful if you run into problems. The log is stored in the Documents\OnlyT\Logs folder.

**Comment by bergerchemical** [comment]
oops! I tested it too briefly, thanks for your help :) (99,45 min)
I will test it again correctly anyway

**Comment by LvDaniel** [comment]
I've tested again and the report is created.
But it gives just the overall status, if the meeting finished in time or not.

[2018-11-14.pdf](https://github.com/AntonyCorbett/OnlyT/files/2578431/2018-11-14.pdf)

**Comment by AntonyCorbett** [comment]
@LvDaniel thanks. That's odd. Did you go through each of the meeting timers?

**Comment by LvDaniel** [comment]
@AntonyCorbett yes, I did.
But I stated the meeting at 11pm and finish it on 12am next day. Also the timer was counting up and at some point I've change it to count down. Maybe those influenced it somehow. 
I'll try once again during the day and I won't do any changes.

**Comment by LvDaniel** [comment]
@AntonyCorbett I've test it once again now the report is working.
[2018-11-14.pdf](https://github.com/AntonyCorbett/OnlyT/files/2580747/2018-11-14.pdf)
Is it possible to have a summary report as well for the previous meetings like in SundBox?
Or this will be created automatically after more reports will be generated?
Thanks.

**Comment by LvDaniel** [comment]
@AntonyCorbett one more question on this topic.
If I use /id=Name for the shortcut, will the report be stored in a separate folder with that Name?

**Comment by AntonyCorbett** [comment]
> Is it possible to have a summary report as...
@LvDaniel yes, it will be printed on a second page as soon as you have 5 or 6 weeks worth of timing data.

**Comment by AntonyCorbett** [comment]
> If I use /id=Name for the shortcut, will the report be stored in a separate folder with that Name?

@LvDaniel Yes
